### PR TITLE
Fix: Not to error when no config path is found

### LINKED ISSUE #874: Running `sonarwhal` when there is not config file results in an TypeError
`sonarwhal` is installed globally, there is no config file, and the first thing that is run is:

```bash
sonarwhal https://example.com
```

Generating:

```bash
TypeError: Path must be a string. Received null
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:1186:7)
    at Function.fromFilePath (/Users/alrra/.nvm/versions/node/v8.9.3/lib/node_modules/sonarwhal/dist/src/lib/config.js:176:35)
    at tryToLoadConfig (/Users/alrra/.nvm/versions/node/v8.9.3/lib/node_modules/sonarwhal/dist/src/lib/cli/analyze.js:49:43)
    at exports.analyze (/Users/alrra/.nvm/versions/node/v8.9.3/lib/node_modules/sonarwhal/dist/src/lib/cli/analyze.js:95:26)
    at Object.exports.execute (/Users/alrra/.nvm/versions/node/v8.9.3/lib/node_modules/sonarwhal/dist/src/lib/cli.js:32:29)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:678:11)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
Couldn't load a valid configuration file in null.
? A valid configuration file can't be found. Do you want to create a new
 one? (Y/n) 
```

### PR DESCRIPTION
**Pull request by qzhou1607**
## Pull request checklist

Make sure you:

- [X] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [X] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messagess)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [X] Added/Updated related tests.

## Short description of the change(s)
Fix #874

### CODE REVIEW

File: packages/sonarwhal/src/lib/cli/analyze.ts [hunk 1]
```diff
@@ -63,6 +63,18 @@ const tryToLoadConfig = async (actions: CLIOptions): Promise<SonarwhalConfig> =>
     let config: SonarwhalConfig;
     const configPath: string = actions.config || SonarwhalConfig.getFilenameForDirectory(process.cwd());
 
+    if (!configPath) {
+        logger.log(`Couldn't find a valid path to load the configuration file.`)
```
Comment (molant) [inline packages/sonarwhal/src/lib/cli/analyze.ts]: Change to `logger.error`

Comment (qzhou1607) [reply packages/sonarwhal/src/lib/cli/analyze.ts]: @molant Done.

File: packages/sonarwhal/src/lib/cli/analyze.ts [hunk 2]
```diff
@@ -63,6 +63,18 @@ const tryToLoadConfig = async (actions: CLIOptions): Promise<SonarwhalConfig> =>
...
+    if (!configPath) {
+        logger.error(`Couldn't find a valid path to load the configuration file.`)
+
+        const created = await askUserToCreateConfig();
+
+        if (created) {
+            config = await tryToLoadConfig(actions);
+        }
+
+        return config;
```
Comment (sarvaje) [inline packages/sonarwhal/src/lib/cli/analyze.ts]: Is this return ok? I mean, do we want to finish the process after create the config or we want to use that config to run sonarwhal?

Comment (molant) [reply packages/sonarwhal/src/lib/cli/analyze.ts]: We want to use that config to run sonarwhal

Comment (qzhou1607) [reply packages/sonarwhal/src/lib/cli/analyze.ts]: @sarvaje This return should be okay. tryToLoadConfig is only responsible for getting config but not running sonarwhal. Even though we return here, sonarwhal will still run later in the analyze function.

Comment (qzhou1607) [reply packages/sonarwhal/src/lib/cli/analyze.ts]: @molant This is `return` in `tryToLoadConfig` but not `analyze`. So sonarwhal should still run after this.

Comment (sarvaje) [reply packages/sonarwhal/src/lib/cli/analyze.ts]: ahhh, I thought it was a return in `analyze`.

**Pull request closed without merging**